public/tracker/modules/task/directives/taskHistory/taskHistory.js

```diff
@@ -0,0 +1,84 @@
+angular
+    .module('Tracker')
+
+    .directive('taskHistory', function () {
+            return {
+                restrict: 'A',
+                templateUrl: 'tracker/modules/task/directives/taskHistory/taskHistory.html',
+                controller: function ($scope,
+                                      Task,
+                                      TaskHistory,
+                                      historyInfoService) {
+                    var allMessages;
+
+                    $scope.getInclude = function (messageType) {
+                        var historyType = _.find(historyInfoService, function (historyType) {
```

Comment:
looks like if we move all rendering to templates we no need historyInfoService, and this service can be remove